### ISSUE TITLE
Compilation fails on alpine Linux due to curl version

### ISSUE THREAD

**Issue by lpil** [appsignal/appsignal-elixir#321]
Output:
```
Generated phoenix app

18:38:17.192 [info]  Downloading agent release from https://appsignal-agent-releases.global.ssl.fastly.net/ee63235
/appsignal-x86_64-linux-musl-all-static.tar.gz
curl: (48) An unknown option was passed in to libcurl
could not compile dependency :appsignal, "mix compile" failed. You can recompile this dependency with "mix deps.co
mpile appsignal", update it with "mix deps.update appsignal" or clean it with "mix deps.clean appsignal"
** (Mix) Download failed with code 48

==> appsignal
==> fcat
The command '/bin/sh -c mix do deps.get --only $MIX_ENV, deps.compile' returned a non-zero code: 1
```

Here's a mix file and dockerfile to replicate the problem:

```elixir
defmodule Fcat.Mixfile do
  use Mix.Project

  def project do
    [
      app: :fcat,
      version: "0.0.1",
      elixir: "~> 1.4",
      start_permanent: Mix.env() == :prod,
      deps: deps()
    ]
  end

  def application do
    [
      mod: {Fcat.Application, []},
      extra_applications: [:logger, :runtime_tools]
    ]
  end

 defp deps do
    [
      # Performance monitoring
      {:appsignal, "~> 1.0"}
    ]
  end
end
```

```dockerfile
FROM bitwalker/alpine-elixir-phoenix:1.6.4

ENV MIX_ENV=prod
RUN apk add --update alpine-sdk coreutils curl

RUN mix do local.hex --force, local.rebar --force
COPY mix.exs mix.lock ./
COPY config/ config/

RUN mix do deps.get --only $MIX_ENV, deps.compile
```

**Comment by stunthamster** [comment]
For now, adding apk add curl-dev will make this work - I'm unsure what the underlying issue is.

**Comment by lpil** [comment]
Thanks hamster person :D

**Comment by jeffkreeftmeijer** [comment]
Hey @lpil and @stunthamster,

Thanks for reporting this issue, and thanks for sharing the workaround! We've been meaning to switch the downloading logic from cURL and using HTTPoison instead [for a while](https://github.com/appsignal/appsignal-elixir/issues/238). We'll look at retrying requests (like we do using cURL's `--retry` flag now) using HTTPoison, and we're hoping to release a fix for this soon.

**Comment by lpil** [comment]
Great, thanks for the update

**Comment by jeffkreeftmeijer** [comment]
Hey @lpil and @stunthamster,

We've [released version 1.7 of the Elixir integration](https://blog.appsignal.com/2018/07/11/elixir-package-1.7.html), which drops the cURL and HTTPoison dependencies and uses hackney to download the agent and communicate with the AppSignal API.

As always; please let us know if you run into any issues running 1.7, and thanks again for reporting this. :)
